eng/Versions.props

```diff
@@ -52,8 +52,7 @@
     <MicrosoftCodeAnalysisCSharpCodeStyleVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
     <MicrosoftCodeAnalysisCSharpVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpVersion>
     <MicrosoftCodeAnalysisNetAnalyzersVersion>7.0.0-preview1.21572.6</MicrosoftCodeAnalysisNetAnalyzersVersion>
-    <!-- Pin compiler version to workaround issue: https://github.com/dotnet/runtime/issues/59908 -->
```

Comment:
nit: should we keep this comment since we are still pinning the version? Afaik, either we don't have `MicrosoftNetCompilersToolsetVersion` or have both , the comment and the property.

Reply:
yeah we should probably remove the pinning now that the underlying issue was fixed?

Reply:
Yeah, I thought this was removed a while ago when I consumed the fix. We should unpin the compiler and just use the arcade's version.